### PR TITLE
[JENKINS-27505] Preserve an empty line at the beginning of a textarea (using Groovy)

### PR DESCRIPTION
**Pull request by ikedam**
[JENKINS-27505](https://issues.jenkins-ci.org/browse/JENKINS-27505)
An alternate way for #1612.

Replaces textarea.jelly to textarea.groovy.
That makes it easy to test whether the first line is an empty line.

This change might cause regressions as existing tests doesn't test all the features of textarea.jelly.
#1612 may be safer.

### CODE REVIEW

File: core/src/main/resources/lib/form/textarea.groovy [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+textarea (id: attrs.id, style: attrs.style,
+    name: name,
+    class: "setting-input ${attrs.checkUrl!=null?'validated':''} ${attrs['codemirror-mode']!=null?'codemirror':''} ${attrs.class}",
+    checkUrl: attrs.checkUrl, checkDependsOn: attrs.checkDependsOn, checkMethod: attrs.checkMethod,
+    rows: h.determineRows(value),
+    readonly: attrs.readonly,
+    "codemirror-mode": attrs["codemirror-mode"],
+    "codemirror-config": attrs["codemirror-config"]
+) {
+  if (StringUtils.startsWith(value, "\r") || StringUtils.startsWith(value, "\n")) {
```
Comment (oleg-nenashev) [inline core/src/main/resources/lib/form/textarea.groovy]: and what about two line breaks in a row?

**Comment by oleg-nenashev** [comment]
The migration to Groovy is OK for me. BTW it could be an overkill for the patch

File: core/src/main/resources/lib/form/textarea.groovy (hunk 1 above)
Comment (ikedam) [inline core/src/main/resources/lib/form/textarea.groovy]: It works.
Only the first line break is ignored.
Please have a look at T3 in `testText`.

**Comment by oleg-nenashev** [comment]
Looks good to me in general. :+1:  
Is there a risk that somebody explicitly includes jelly file in a groovy-incompatible style?

**Comment by oleg-nenashev** [comment]
#1612 has been merged, this one is an improvement.
@jenkinsci/core Would we like to go forward with the replacement to groovy?

**Comment by ikedam** [comment]
#1612 was merged instead.
Thanks.

**Pull request closed without merging**